File: dayTwo/python/caesar_cipher.py
def get_encrypted_cipher_value(character, shift_value):
    ascii_value = ord(character)
    new_character_ascii_value = ascii_value + shift_value

    if character.isupper() and new_character_ascii_value > 90:
        remainder = new_character_ascii_value - 90
        new_character_ascii_value = 65 + remainder - 1

    if character.islower() and new_character_ascii_value > 122:
        remainder = new_character_ascii_value - 122
        new_character_ascii_value = 97 + remainder - 1

    return chr(new_character_ascii_value)


def get_decrypted_cipher_value(character, shift_value):
    ascii_value = ord(character)
    new_character_ascii_value = ascii_value - shift_value

    if character.isupper() and new_character_ascii_value < 65:
        remainder = 65 - new_character_ascii_value
        new_character_ascii_value = 90 - remainder + 1

    if character.islower() and new_character_ascii_value < 97:
        remainder = 97 - new_character_ascii_value
        new_character_ascii_value = 122 - remainder + 1

    return chr(new_character_ascii_value)


def get_encrypted_text(message, shift_value):
    encrypted_message = ""

    for character in message:
        if character.isalpha():
            encrypted_message += get_encrypted_cipher_value(character, shift_value)
        else:
            encrypted_message += character

    return encrypted_message


def get_decrypted_text(message, shift_value):
    decrypted_message = ""

    for character in message:
        if character.isalpha():
            decrypted_message += get_decrypted_cipher_value(character, shift_value)
        else:
            decrypted_message += character

    return decrypted_message

File: dayTwo/java/CaesarCipher.java
public class CaesarCipher {

    public static char getEncryptedCipherValue(char character, int shiftValue) {
        int asciiValue = (int) character;
        int newCharacterAsciiValue = asciiValue + shiftValue;

        if (Character.isUpperCase(character) && newCharacterAsciiValue > 90) {
            int remainder = newCharacterAsciiValue - 90;
            newCharacterAsciiValue = 65 + remainder - 1;
        }

        if (Character.isLowerCase(character) && newCharacterAsciiValue > 122) {
            int remainder = newCharacterAsciiValue - 122;
            newCharacterAsciiValue = 97 + remainder - 1;
        }

        return (char) newCharacterAsciiValue;
    }

    public static char getDecryptedCipherValue(char character, int shiftValue) {
        int asciiValue = (int) character;
        int newCharacterAsciiValue = asciiValue - shiftValue;

        if (Character.isUpperCase(character) && newCharacterAsciiValue < 65) {
            int remainder = 65 - newCharacterAsciiValue;
            newCharacterAsciiValue = 90 - remainder + 1;
        }

        if (Character.isLowerCase(character) && newCharacterAsciiValue < 97) {
            int remainder = 97 - newCharacterAsciiValue;
            newCharacterAsciiValue = 122 - remainder + 1;
        }

        return (char) newCharacterAsciiValue;
    }

    public static String getEncryptedTest(String message, int shiftValue) {
        String encryptedMessage = "";

        for (int index = 0; index < message.length(); index++) {
            char character = message.charAt(index);

            if (Character.isLetter(character))
                encryptedMessage += getEncryptedCipherValue(character, shiftValue);

            else
                encryptedMessage += character;
        }

        return encryptedMessage;
    }

    public static String getDecryptedText(String message, int shiftValue) {
        String decryptedMessage = "";

        for (int index = 0; index < message.length(); index++) {
            char character = message.charAt(index);

            if (Character.isLetter(character))
                decryptedMessage += getDecryptedCipherValue(character, shiftValue);

            else
                decryptedMessage += character;
        }

        return decryptedMessage;
    }
}
